packages/indexer-common/src/indexer-management/allocations.ts

```diff
@@ -1131,27 +1133,36 @@ export class AllocationManager {
       allocates = BigNumber.from(0)
     }
 
-    // Handle unallocations
+    // Handle unallocations.
+    // We intentionally don't check if the allocation is active now because it will be checked
+    // later, when we prepare the transaction.
+
     if (action.type === ActionType.UNALLOCATE || action.type === ActionType.REALLOCATE) {
-      // ensure this Action have a valid allocationID
+      // Ensure this Action have a valid allocationID
       if (action.allocationID === null || action.allocationID === undefined) {
         throw Error(
           `SHOULD BE UNREACHABLE: Unallocate or Reallocate action must have an allocationID field: ${action}`,
         )
       }
+
       // Fetch the allocation on chain to inspect its amount
       const allocation = await this.networkMonitor.allocation(action.allocationID)
 
-      /* We intentionally don't check if the allocation is active now because it will be checked
-       * later when we prepare the transaction */
+      // Accrue rewards, except for null or zeroed POI
+      rewards =
+        !action.poi || action.poi === '0' || action.poi === '0x0'
```

Comment:
When comparing against a "zero" POI I'd suggest using `hexlify()` to generate the comparison value. 

Like: `action.poi === utils.hexlify(Array(32).fill(0))`

Reply:
Fixed in c8b6254